File: src/main/java/vergisst/minecraftmod/weaponthrow/enchantments/ThrowEnchantment.java
package vergisst.minecraftmod.weaponthrow.enchantments;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.enchantment.EnchantmentTarget;
import net.minecraft.entity.EquipmentSlot;
import net.minecraft.item.AxeItem;
import net.minecraft.item.ItemStack;
import vergisst.minecraftmod.weaponthrow.handler.ConfigRegistry;

public class ThrowEnchantment extends Enchantment {

    public ThrowEnchantment(Enchantment.Rarity rarityIn, EquipmentSlot... slots) {
        super(rarityIn, EnchantmentTarget.WEAPON, slots);
    }
    // Returns the minimal value of enchantability needed on the enchantment level passed.
    public int getMinPower(int enchantmentLevel) {
        return 10;
    }
    public int getMaxPower(int enchantmentLevel) {
        return this.getMinPower(enchantmentLevel) + 40;
    }
    // Returns the maximum level that the enchantment can have.
    public int getMaxLevel() {
        return 3;
    }

    @Override
    public boolean isAcceptableItem(ItemStack stack) {
        boolean enchantAll = ConfigRegistry.COMMON.getConfig().enchantments.enchantAllWeapons;
        boolean isAxe = stack.getItem() instanceof AxeItem;
        boolean canApply = super.isAcceptableItem(stack);
        return (isAxe || canApply || enchantAll) && ConfigRegistry.COMMON.getConfig().enchantments.enableThrow;
    }
}

File: src/main/kotlin/vergisst/minecraftmod/weaponthrowlite/enchantment/ThrowEnchantment.kt
package vergisst.minecraftmod.weaponthrowlite.enchantment

import net.minecraft.enchantment.Enchantment
import net.minecraft.enchantment.EnchantmentTarget
import net.minecraft.entity.EquipmentSlot
import net.minecraft.item.AxeItem
import net.minecraft.item.ItemStack
import vergisst.minecraftmod.weaponthrowlite.handlers.ConfigRegistry

class ThrowEnchantment(rarity: Rarity, vararg slots: EquipmentSlot): Enchantment(rarity, EnchantmentTarget.WEAPON, slots) {

    override fun getMinPower(_a: Int) = 10
    override fun getMaxPower(_a: Int) = 50

    override fun getMaxLevel() = 3

    override fun isAcceptableItem(stack: ItemStack): Boolean {
        val enchantAll: Boolean = ConfigRegistry.COMMON.getConfig().enchantments.enchantAllWeapons
        val isAxe = stack.item is AxeItem
        val canApply = super.isAcceptableItem(stack)
        return (isAxe || canApply || enchantAll) && ConfigRegistry.COMMON.getConfig().enchantments.enableThrow
    }
}